flow/designs/asap7/mock-array-big/README.md

```diff
@@ -33,3 +33,63 @@ Now build the design as usual:
 ```
 make
 ```
+
+Element/constraint.sdc
+----------------------
+
+A single macro is made for all the elements in the array, though strictly
+speaking each macro is unique in that the timing constraints of each element
+in the array is different.
+
+To illustrate, consider the `assign io_lsbOuts_0 = io_lsbIns_1;` statement in Element.v.
+Here a signal is routed from left to right, through the Element without the signal
+being registered.
+
+Clearly, the maximum input delay for `io_lsbIns_1` is
+smaller for the leftmost than for the rightmost Element.
+
+Taking a step back, it is also worth considering what level of detail is
+appropriate for the Element's constraint.sdc file.
+
+In the beginning of a project, during exploration, the .sdc file does not
+need to be particularly detailed. In fact, details can be counterproductive.
+
+Consider minimum input delay, which relates to hold times. If the main concern
+of architectural exploration is to ensure that the design can operate on a
+sufficiently high frequency, then specifying minimum input delays during
+exploration is premature. Minimum input delay is a constraint that belongs at
+the end the development cycle when the design is well established and won't change
+and the concern is to lock down the macro for it's specific operating frequency.
+
+Choosing a strategy for Element/constraint.sdc combinational paths
+------------------------------------------------------------------
+
+It isn't possible to articulate a single .sdc file that exactly captures the
+constraints of an Element as each element is used in unique circumstances.
+Instead, a choice has to made and each choice has it's pros and cons.
```

Comment:
```suggestion
Instead, a choice has to made and each choice has its pros and cons.
```

Reply:
what are the cons against using a unique SDC file for each element ?